src/stepfunctions/steps/states.py

```diff
@@ -583,14 +583,21 @@ def __init__(self, state_id, **kwargs):
             state_id (str): State name whose length **must be** less than or equal to 128 unicode characters. State names **must be** unique within the scope of the whole state machine.
             resource (str): A URI that uniquely identifies the specific task to execute. The States language does not constrain the URI scheme nor any other part of the URI.
             timeout_seconds (int, optional): Positive integer specifying timeout for the state in seconds. If the state runs longer than the specified timeout, then the interpreter fails the state with a `States.Timeout` Error Name. (default: 60)
+            timeout_seconds_path (str, optional): Path applied to the state's input to select the integer to be used for timeout seconds duration.
```

Comment:
Grammar and extra help taken from the [docs](https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-task-state.html):

```suggestion
            timeout_seconds_path (str, optional): Path specifying the state's timeout value in seconds from the state input. When resolved, the path must select a field whose value is a positive integer.
```

Same suggestion applies for `heartbeat_seconds_path`